src/validation.cpp

```diff
@@ -3773,7 +3773,9 @@ bool BlockManager::LoadBlockIndex(
             pindex->nStatus |= BLOCK_FAILED_CHILD;
             setDirtyBlockIndex.insert(pindex);
         }
-        if (pindex->IsValid(BLOCK_VALID_TRANSACTIONS) && (pindex->HaveTxsDownloaded() || pindex->pprev == nullptr)) {
+        if (pindex->IsAssumedValid() ||
```

Comment:
I don't totally understand your comment here. This is a necessary change so that when loading a snapshot-based chainstate, the (assumed-valid) tip is able to be added as a tip candidate, since otherwise it would be excluded on the basis of `!HaveTxsDownloaded()`.

Reply:
I had the wrong mental model reviewing. I thought that only blocks before the _snapshot_ height were marked as `IsAssumeValid`, thanks for the clarification!